### PR TITLE
[18965] Improved threading model information

### PR DESCRIPTION
**Pull request by jsantiago-eProsima**
Partial rewriting of threading information to include additional information and all threads spawned by Fast DDS, as well as their cardinality and conditions.

### CODE REVIEW

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 1]
```diff
@@ -15,14 +15,69 @@ any methods on the same DomainParticipant from different threads.
...
+      - Accumulates and writes to the appropriate consumer log entries.
+    * - Security Logging
+      - Log
+      - One
+      - Accumulates and writes security log entries.
+
+Some of these threads are only spawned when certain conditions are met.
+Datasharing listener thread is created only when Datasharing is in use.
+TCP keep alive thread requires the keep alive period to be configured to a value greater than zero.
+Security logging and Shared Memory packet logging threads both require certain configuration options to be enabled.
```
Comment (EduPonz) [inline docs/fastdds/library_overview/includes/exec_model.rst:80]: I'd put this as a list here

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 2]
```diff
@@ -15,14 +15,69 @@ any methods on the same DomainParticipant from different threads.
...
+      - Manages asynchronous writes.
+
+        Even for synchronous writers, some forms of communication must be initiated in the background.
+    * - Datasharing Listener
+      - General
+      - One per DomainParticipant
+      - Listener thread that processes messages received via Datasharing
+    * - Reception
+      - UDP
+      - One per port
```
Comment (EduPonz) [inline docs/fastdds/library_overview/includes/exec_model.rst:46]: Either we link to the default ports somehow (IDK whether we have that) or we elabore further here on the number of ports

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 3]
```diff
@@ -15,14 +15,69 @@ any methods on the same DomainParticipant from different threads.
...
+      - One
+      - Monitors health of open shared memory segments.
+    * - General Logging
+      - Log
+      - One
+      - Accumulates and writes to the appropriate consumer log entries.
+    * - Security Logging
+      - Log
+      - One
+      - Accumulates and writes security log entries.
```
Comment (EduPonz) [inline docs/fastdds/library_overview/includes/exec_model.rst:75]: Missing:

1. The FileWatch threads
1. Discovery Server thread

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 4]
```diff
@@ -15,14 +15,69 @@ any methods on the same DomainParticipant from different threads.
...
+    * - Event
+      - General
+      - One per DomainParticipant
+      - Processes periodic and triggered time events
+    * - Asynchronous Writer
+      - General
+      - One per enabled asynchronous flow controller
+      - Manages asynchronous writes.
+
+        Even for synchronous writers, some forms of communication must be initiated in the background.
```
Comment (EduPonz) [inline docs/fastdds/library_overview/includes/exec_model.rst:39]: I think it's worth stating that by default there is one of these

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/fastdds/library_overview/includes/exec_model.rst [hunk 5]
```diff
@@ -15,14 +15,92 @@ any methods on the same DomainParticipant from different threads.
...
-  depends on the transport layer (e.g. a UDP port).
+
+The complete set of threads spawned by Fast DDS is shown below.
+Transport related threads (marked as UDP, TCP and SHM types) are only created when the appropriate Transport is used.
+
+.. list-table::
+    :header-rows: 1
+    :align: left
+
+    * - Name
```
Comment (rsanchez15) [inline docs/fastdds/library_overview/includes/exec_model.rst:26]: I'd try to fit the table into the page width

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: docs/fastdds/library_overview/includes/exec_model.rst (hunk 2 above)
Comment (EduPonz) [reply docs/fastdds/library_overview/includes/exec_model.rst:54]: I don't think this has been addressed

Comment (EduPonz) [reply docs/fastdds/library_overview/includes/exec_model.rst:54]: Nevermind

**Review by EduPonz — DISMISSED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**